src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -348,6 +374,8 @@ class CORE_EXPORT QgsSymbolLegendNode : public QgsLayerTreeModelLegendNode
   private:
     void updateLabel();
 
+    QString injectSymbolId( const QString &expression );
```

Comment:
unused?

Reply:
No this was to be the class to do parsing but as said in my latest comment I think it is wiser to give that responsibility to the user.